SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -539,15 +539,22 @@ void USpatialSender::SendComponentUpdates(UObject* Object, const FClassInfo& Inf
 }
 
 // Apply (and clean up) any updates queued, due to being sent previously when they didn't have authority.
-void USpatialSender::ProcessUpdatesQueuedUntilAuthority(Worker_EntityId EntityId)
+void USpatialSender::ProcessUpdatesQueuedUntilAuthority(Worker_EntityId EntityId, Worker_ComponentId ComponentId)
 {
 	if (TArray<Worker_ComponentUpdate>* UpdatesQueuedUntilAuthority = UpdatesQueuedUntilAuthorityMap.Find(EntityId))
 	{
-		for (Worker_ComponentUpdate& Update : *UpdatesQueuedUntilAuthority)
+		for (auto It = UpdatesQueuedUntilAuthority->CreateIterator(); It; It++)
+		{
+			if (ComponentId == It->component_id)
+			{
+				Connection->SendComponentUpdate(EntityId, &(*It));
+				It.RemoveCurrent();
```

Comment:
No memory corruption?

Reply:
`RemoveCurrentSwap`

Reply:
Have had to replace the entire code section, now it's kind of less readable. Suggestions welcome

Reply:
Replaced back to using an iterator, because we want to preserve order.